srml/staking/src/lib.rs

```diff
@@ -1187,7 +1188,7 @@ impl<T: Trait> Module<T> {
 
 			for (v, p) in validators.iter().zip(points.individual.into_iter()) {
 				if p != 0 {
-					let reward = multiply_by_rational(total_payout, p, points.total);
+					let reward = Perbill::from_rational_approximation(p, points.total) * total_payout;
```

Comment:
I'm not sure introducing Per-u32 would solve it what I had in mind was this:
```
p = 1, points.total = 3, total_payout = 3000).
with Per-thing it will do
3000*333_333_333/1_000_000_000 which will returns 999.
```

possibilities:
* maybe introducing rational(u32, u32) would solve this rounding error.
* maybe we can actually say that this is fine
* also we really care about this rounding error we could do rounding in the multiplication of per-thing instead of just a cut operation with something like this:
```diff
diff --git a/core/sr-primitives/src/lib.rs b/core/sr-primitives/src/lib.rs
index f6784c8f2..90019188e 100644
--- a/core/sr-primitives/src/lib.rs
+++ b/core/sr-primitives/src/lib.rs
@@ -213,7 +213,11 @@ where
 
                        // `rem_multiplied_u64` is less than 10^12 therefore divided by a million it fits into
                        // u32
-                       let rem_multiplied_divided_u32 = (rem_multiplied_u64 / 1_000_000) as u32;
+                       let mut rem_multiplied_divided_u32 = (rem_multiplied_u64 / 1_000_000) as u32;
+
+                       if rem_multiplied_u64 % 1_000_000 >= 1_000_000/2 {
+                               rem_multiplied_divided_u32 += 1;
+                       }
 
                        // `rem_multiplied_divided` is inferior to b, thus it can be converted back to N type
                        rem_multiplied_divided_u32.into()
```

Reply:
I think in this case since the difference of 1billion and 4 billion is not that much, we can opt to `maybe we can actually say that this is fine`

I will also add your suggestion. sounds reasonable.

Reply:
applied